libopenage/pathfinding/grid.cpp

```diff
@@ -64,6 +64,39 @@ const std::vector<std::shared_ptr<Sector>> &Grid::get_sectors() const {
 	return this->sectors;
 }
 
+
+const Grid::portalgraph_t &Grid::get_portal_graph() const {
+	return this->portal_graph;
+}
+
+void Grid::init_portal_graph()
+{
+	//map
+	for(auto& sector: this->get_sectors())
+	{
+		for(auto& portal : sector->get_portals())
+		{
+			//node_t portal_node = std::make_shared<PortalNode>(portal, sector->get_id(), nullptr);
+			std::map<portal_id_t, int> links = this->portal_graph[portal->get_id()];
+			for(auto& exit : portal->get_exits(sector->get_id()))
+			{
+				//node_t exit_node = std::make_shared<PortalNode>(exit, portal->get_exit_sector(portal_node->entry_sector), nullptr);
+				int cost = Pathfinder::distance_cost(
+								portal->get_exit_center(sector->get_id()),
+								exit->get_entry_center(portal->get_exit_sector(sector->get_id()))); 
```

Comment:
Precomputing the distance cost is a pretty good idea! However, the distance cost should probably be stored in the `PortalNode` class.

Reply:
@heinezen hey thanks for the all the advice, I will try to address the comments.